src/core/unit.c

```diff
@@ -1278,19 +1285,19 @@ int unit_add_exec_dependencies(Unit *u, ExecContext *c) {
         /* Unlike unit_add_dependency() or friends, this always returns 0 on success. */
 
         if (c->working_directory && !c->working_directory_missing_ok) {
-                r = unit_require_mounts_for(u, c->working_directory, UNIT_DEPENDENCY_FILE);
+                r = unit_add_mounts_for(u, c->working_directory, UNIT_DEPENDENCY_FILE, UNIT_MOUNT_REQUIRES);
```

Comment:
hmm, plesae drop the `!c->working_directory_missing_ok` from the outer `if` condition here, and instead make the last argument of the call a ternary op: `c->working_directory_missing_ok ? UNIT_MOUNT_WANTS : UNIT_MOUNT_REQUIRES`.